src/WholeBodyStateDisplay.cpp

```diff
@@ -588,28 +653,35 @@ void WholeBodyStateDisplay::processWholeBodyState() {
   size_t n_suppcontacts = 0;
   grf_visual_.clear();
   cones_visual_.clear();
-  Eigen::Vector3d cop_pos = Eigen::Vector3d::Zero();
+  cop_visual_.clear();
+  cop_pos_.clear();
+  Eigen::Vector3d zmp_pos = Eigen::Vector3d::Zero();
   Eigen::Vector3d total_force = Eigen::Vector3d::Zero();
   for (size_t i = 0; i < num_contacts; ++i) {
     const whole_body_state_msgs::ContactState &contact = msg_->contacts[i];
     std::string name = contact.name;
 
-    // Getting the contact position
+    // Getting the contact position and orientation
     Ogre::Vector3 contact_pos(contact.pose.position.x, contact.pose.position.y, contact.pose.position.z);
+    Eigen::Vector3d contact_ref_dir = -Eigen::Vector3d::UnitY();
+    Eigen::Vector3d contact_dir(contact.surface_normal.x, contact.surface_normal.y, contact.surface_normal.z);
+    Eigen::Quaterniond contact_q;
+    contact_q.setFromTwoVectors(contact_ref_dir, contact_dir);
+    Ogre::Quaternion contact_orientation(contact_q.w(), contact_q.x(), contact_q.y(), contact_q.z());
 
```

Comment:
@cmastalli Here is where I get the contact surface orientation. I pretty much copied this from the friction cone code with appropriate name changes.

Reply:
This looks alright, but it would be nice if you can double-check.

Can you run a simulation with an inclined surface? If so, could you share the video?